src/test/calculator/proportionalMatch.test.ts

```diff
@@ -0,0 +1,196 @@
+
+class FakePassportProvider implements PassportProvider {
+  scores: {
+    [address: string]: PassportScore;
+  };
+
+  constructor(scores: PassportScore[]) {
+    this.scores = {};
+    for (const s of scores) {
+      this.scores[s.address] = s;
+    }
+  }
+
+  start(_opts?: { watch: boolean } | undefined) {
+    return Promise.resolve(undefined);
+  }
+
+  stop(): void {}
+
+  async getScoreByAddress(address: string) {
+    return this.scores[address];
+  }
+}
+
+const round: Round = {
+  id: "0x1234",
+  amountUSD: 0,
+  votes: 0,
+  token: "0x1234",
+  matchAmount: "0x0",
+  matchAmountUSD: 0,
+  uniqueContributors: 0,
+  applicationMetaPtr: "",
+  applicationMetadata: null,
+  metaPtr: "",
+  metadata: null,
+  applicationsStartTime: "",
+  applicationsEndTime: "",
+  roundStartTime: "",
+  roundEndTime: "",
+  createdAtBlock: 0,
+  updatedAtBlock: 0,
+};
+
+const applications: Application[] = [
+  {
+    id: "application-id-1",
+    projectId: "project-id-1",
+    roundId: "0x1234",
+    status: "APPROVED",
+    amountUSD: 0,
+    votes: 0,
+    uniqueContributors: 0,
+    metadata: {
+      application: {
+        project: {
+          title: "",
+          website: "",
+          projectTwitter: "",
+          projectGithub: "",
+          userGithub: "",
+        },
+        answers: [],
+        recipient: "grant-address-1",
+      },
+    },
+    createdAtBlock: 0,
+    statusUpdatedAtBlock: 0,
+    statusSnapshots: [
+      {
+        status: "APPROVED",
+        statusUpdatedAtBlock: 0,
+      },
+    ],
+  },
+];
+
+const MOCK_CHAIN = {
+  id: 250,
+  tokens: [
+    {
+      code: "GcV",
+      address: "0x83791638da5EB2fAa432aff1c65fbA47c5D29510",
+      voteAmountCap: BigInt(10e18),
+    },
+    {
+      code: "Dummy",
+      address: "0x1234",
+    },
+  ],
+} as unknown as Chain;
+
+let voteId = 0;
+
+function generateVoteAndScore(id: number, amount: bigint, rawScore: string) {
+  const vote = {
+    id: `vote-${id}`,
+    projectId: "project-id-1",
+    applicationId: "application-id-1",
+    roundId: "round-id-1",
+    token: "0x83791638da5EB2fAa432aff1c65fbA47c5D29510",
+    voter: `voter-${id}`,
+    grantAddress: "grant-address-1",
+    amount: amount.toString(),
+    amountUSD: Number(amount),
+    amountRoundToken: amount.toString(),
+  };
+
+  const score = {
+    address: `voter-${id}`,
+    score: "xyz",
+    status: "XYZ",
+    last_score_timestamp: "2023-05-08T10:17:52.872812+00:00",
+    evidence: {
+      type: "ThresholdScoreCheck",
+      // success is not used anymore
+      success: false,
+      // rawScore is the only attribute used
+      rawScore,
+      // threshold is not used
+      threshold: "15.00000",
+    },
+    error: null,
+  };
+
+  return { vote, score };
+}
+
+describe("getVotesWithCoefficients", () => {
+  let votes: Vote[] = [];
+  let scores: PassportScore[] = [];
+  let fakePassportProvider: FakePassportProvider;
+
+  beforeAll(() => {
```

Comment:
If you want to get fancy / extra tidy here, vitest has a "test context" that works great for separating fixture code from test code: https://vitest.dev/guide/test-context.html

There's an example of that in the passport provider tests.